atom/common/node_bindings.cc

```diff
@@ -173,6 +175,18 @@ node::Environment* NodeBindings::CreateEnvironment(
       command_line->argv();
 #endif
 
+  if (is_browser_ && AtomCommandLine::InitialisedForCurrentProcess()) {
+    auto new_args = AtomCommandLine::ForCurrentProcess()->argv();
+    auto it = std::find(args.begin(), args.end(),
+                        "--atom-shell-switches-start");
+    for (; *it != "--atom-shell-switches-end"; it++) {
+      new_args.push_back(*it);
+    }
+    new_args.insert(new_args.end(), *it);
+    args.swap(new_args);
+    AtomCommandLine::Reset();
+  }
```

Comment:
@zcbenz then this hack would be present to stitch atom-shell swtich arguments from https://github.com/atom/atom-shell/blob/master/atom/app/atom_main_delegate.cc#L71 before sending to `node::Environment` ?